src/System.Net.Sockets/tests/FunctionalTests/SendReceive.cs

```diff
@@ -1133,6 +1133,73 @@ public void SocketSendWouldBlock_ReturnsBytesSent()
                 }
             }
         }
+
+        [Fact]
+        [PlatformSpecific(TestPlatforms.AnyUnix)]
+        [ActiveIssue(31766, TestPlatforms.OSX)]
+        public async Task Socket_ReceiveFlags_Success()
+        {
+            using (var sender = new Socket(AddressFamily.InterNetwork, SocketType.Dgram, ProtocolType.Udp))
+            using (var receiver = new Socket(AddressFamily.InterNetwork, SocketType.Dgram, ProtocolType.Udp))
+            {
+                receiver.BindToAnonymousPort(IPAddress.Loopback);
+                receiver.ReceiveTimeout = TestSettings.PassingTestTimeout;
+                sender.Connect(receiver.LocalEndPoint);
+                sender.SendBufferSize = 1500;
+
+                var data = new byte[500];
+                data[0] = data[499] = 1;
+
+                // Send() can send less than asked for but datagram should be atomic and fit to send buffer.
+                Assert.Equal(500, sender.Send(data));
```

Comment:
Not understanding this comment. Is it saying that while sometimes Send will send less than asked for, in this case it's guaranteed to send it all?

Reply:
I look and it seems like the short read/write can happen only on non-blocking socket in c#. 
So I remove the comment. The Send() should not block anyway as I set socket buffer big enough to fit in both messages.